sharing_portal/views.py

```diff
@@ -558,4 +863,39 @@ def _request_artifact_dois(artifact, request_forms=[]):
             return True
         return False
     except Exception:
-        LOG.exception('Failed to request DOI for artifact {}'.format(artifact.pk))
+        LOG.exception("Failed to request DOI for artifact {}".format(artifact.pk))
+
+
+def create_supplemental_project(request, artifact):
+    mapper = ProjectAllocationMapper(request)
+
+    pi = artifact.project.pi
+    supplemental_project = {
+        "nickname": f"reproducing_{artifact.id}",
+        "title": f"Reproducing '{artifact.title}'",
+        "description": f"This project is for reproducing the artifact '{artifact.title}'",
+        "typeId": artifact.project.type.id,
+        "fieldId": artifact.project.field.id,
+        "piId": artifact.project.pi.id,
+    }
+    # Approval code is commented out during initial preview release.
+    allocation_data = {
+        "resourceId": 39,
+        "requestorId": pi.id,
+        "computeRequested": 1000,
+        "status": "approved",
+        #"dateReviewed": timezone.now(),
+        #"start": timezone.now(),
+        #"end": timezone.now() + timedelta(days=6*30),
+        #"decisionSummary": "automatically approved for reproducibility",
+        #"computeAllocated": 1000,
+        #"justification": "Automatic decision",
+    }
+    supplemental_project["allocations"] = [allocation_data]
+    supplemental_project["source"] = "Chameleon"
```

Comment:
I was thinking we should get rid of this "source" thing at some point, but this is actually the first time it may be useful :D -- we could say "Day pass" or something, idk. Probably not worth it.

Reply:
I've changed it to day pass, it is a simple change that probably doesn't hurt anything.